test/integration/test_connection_management.py

```diff
@@ -95,3 +95,39 @@ def test_http_h2_connection_management_single_request_per_conn_1(http_test_serve
 @pytest.mark.skipif(isSanitizerRun(), reason="Unstable in sanitizer runs")
 def test_http_h2_connection_management_single_request_per_conn_1(http_test_server_fixture):
   connection_management_test_request_per_connection(http_test_server_fixture, 5, True)
+
+
+@pytest.mark.skipif(isSanitizerRun(), reason="Unstable in sanitizer runs")
```

Comment:
Why are these unstable? Ideally we would have all inputs hermetic and not have any environmental deps.

Reply:
I backed out the unstable marking of this specific test, let's see how that goes. 
So far the "unstable in sanitizer runs" marking has been limited to the ones related to connection management. 

In general, these end-to-end integration tests are using the binaries we produce and verifying they work as intended. They aren't using simulated time... now the sanitizers sometimes induce excessive processing slowdowns, which can make some of these tests become unstable when they rely on "reasonable" timings.
For example: if we expect a connection to be re-used, excessive reply latency can cause test failures, because the connection doesn't free up in time (before we issue a new request).

I opted for marking them as unstable, but a different solution might be to tune parameters to allow more timing slack; however I suspect that doing so is going to increase test duration considerably overall (unless we start introducing specializations in tests for sanitizer awareness).